src/main/java/wromaciej/hvac_sim/solver/externals/JunctionSolver.java

```diff
@@ -0,0 +1,90 @@
+package wromaciej.hvac_sim.solver.externals;
+
+import org.springframework.stereotype.Service;
+import wromaciej.hvac_sim.solver.result.SolverResult;
+import wromaciej.hvac_sim.solver.result.SolverResultType;
+import wromaciej.hvac_sim.thermo.generals.bonds.BondDirection;
+import wromaciej.hvac_sim.thermo.generals.conservationLaw.Junction;
+import wromaciej.hvac_sim.thermo.generals.conservationLaw.ParameterWithDirection;
+import wromaciej.hvac_sim.thermo.matter.fluids.parameters.Parameter;
+import wromaciej.hvac_sim.thermo.matter.fluids.parameters.ParameterType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class JunctionSolver implements ExternalSolver<Junction> {
+
+
+    public List<ParameterWithDirection> getAllParameters(Junction toSolve) {
+        return toSolve.getAllParameters();
+    }
+
+    public List<ParameterWithDirection> getUndefinedParameters(Junction toSolve) {
+        List<ParameterWithDirection> undefinedParameters = new ArrayList<>();
```

Comment:
use something like:
```
toSolve.getAllParametets().stream().filter(p -> p.getParameter().isDefined()).collect(toList());
```
I am writing from memory so it can be slightly different. generally use stream and java 8 lambda power.

Reply:
in that and many other places